static/src/javascripts/projects/commercial/modules/third-party-tags.ts

```diff
@@ -51,23 +54,29 @@ const addScripts = (tags) => {
 		tag.loaded = true;
 	});
 
+	// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- false positive
 	if (hasScriptsToInsert) {
-		fastdom.mutate(() => {
-			if (ref && ref.parentNode) {
+		void fastdom.mutate(() => {
+			if (ref.parentNode) {
 				ref.parentNode.insertBefore(frag, ref);
 			}
 		});
 	}
 };
 
 const insertScripts = (
-	advertisingServices,
-	performanceServices, // performanceServices always run
-) => {
+	advertisingServices: ThirdPartyTag[],
+	performanceServices: ThirdPartyTag[], // performanceServices always run
+): Promise<void> => {
```

Comment:
Returning void here breaks tests in [third-party-tags.spec.ts](static/src/javascripts/projects/commercial/modules/third-party-tags.spec.ts) which `await` this function then make assertions about the expected behaviour of `addScripts`. 
Interestingly, the tests succeeded in the javascript version.

Reply:
Were the tests really passing in JS, or was it simply not failing? For example, a `Promise` that never settles is not a failure, just a warning in Jest.

Reply:
🤯  that explains it! I wouldn't have expected that behaviour at all.
^ Yes, the tests were actually passing and now we know why